src/PowerShellEditorServices.Hosting/EditorServicesLoader.cs

```diff
@@ -406,5 +408,20 @@ private void ValidateConfiguration()
                 throw new ArgumentNullException(nameof(_hostConfig.PSHost));
             }
         }
+
+        private static object GetPSVersion()
+        {
+#if CoreCLR
+            return typeof(PSObject).Assembly
+                .GetType("System.Management.Automation.PSVersionInfo")
+                .GetMethod("get_PSVersion", BindingFlags.Static | BindingFlags.Public)
+                .Invoke(null, Array.Empty<object>());
+#else
+            return typeof(PSObject).Assembly
+                .GetType("System.Management.Automation.PSVersionInfo", BindingFlags.Instance | BindingFlags.NonPublic)
+                .GetMethod("get_PSVersion", BindingFlags.Static | BindingFlags.NonPublic)
+                .Invoke(null, Array.Empty<object>());
+#endif
+        }
```

Comment:
isn't this in the `VersionUtils` class already?

Reply:
If it it, we can't use it unfortunately, since we can't load PSES.dll at this stage.

Reply:
What about using this?

```csharp
pwsh.Runspace.SessionStateProxy.PSVariable.Get(name);
```

Reply:
Could try that!

Reply:
So, the language mode check works, but the `GetVariable` API doesn't because it throws on concurrent access. At that point we'd need to spin up a new runspace to use it, which wouldn't be good for startup. But I simplified the reflection a bit at least